app/models/rule_group_rule.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+# Join table for the required and conflicting has-many-belongs-to-many relations between
+# RuleGroup and Rule
+class RuleGroupRule < ApplicationRecord
+  belongs_to :requirable, polymorphic: true, optional: true
+  belongs_to :conflictable, polymorphic: true, optional: true
+  belongs_to :selectable, polymorphic: true
+  validates :requirable_id, uniqueness: { scope: %i[selectable_id] }
+  validates :conflictable_id, uniqueness: { scope: %i[selectable_id] }
```

Comment:
Better semantics:
* requirable -> requiring
* conflictable -> conflicting
* selectable -> including

Reply:
I was following the polymorphic association naming conventions for these because I wasn't sure if I was able to stray away from them but I agree that these suggestions sound better. I'll update the names!

Reply:
After googling a little, it does look like we can use whatever name we want and that the 'able' suffix is just to make it clear that it's a polymorphic association, but not necessary.

Reply:
Yup, first of all I should be able to understand it :rofl: pun intended